content/en/docs/Concepts/crds/bundleIndexImage.md

```diff
@@ -0,0 +1,97 @@
+```
+opm index add --bundles <registry>/<username>/sample-operator-bundle:vX.X.X --tag <registry>/<username>/sample-operator-index:vX.X.X --binary-image <registry_base_image>
+```
+- A comma seperated list of additional bundle images to the index
+- The image tag that you want the index image to have.
+- Optional: An alternative registry base image to use for serving the catalog.
+
+2. Push the index image to a registry:
+```
+docker push <registry>/<username>/sample-operator-index:vX.X.X
+```
+
+The index image holds a sqlite database with bundle definitions, it also runs a grpc service when the image is executed. The grpc service lets consumers query the sqlite database for information about the operators this index contains.
+
+You can download the opm command from the [operator-registry](https://github.com/operator-framework/operator-registry) repository.
+
+#### Creating a catalog from an index image
+
+Create a CatalogSource object that references your index image. Modify the following to your specifications and save it as a catalogSource.yaml file:
+
+```yaml
+kind: CatalogSource
+metadata:
+  name: sample-operator
+  namespace: operators
+spec:
+  sourceType: grpc
+  image: <registry>:<port>/<namespace>/sample-operator-index:vX.X.X
+  displayName: My Operator Catalog
+  publisher: <publisher_name> 
+  updateStrategy:
+    registryPoll: 
+      interval: 30m
+```
+
+- Specify your index image
+- Specify your name or an organization name publishing the catalog
+- Catalog sources can automatically check for new versions to keep up to date
+
+Now we have to apply this yaml file to the kubernetes cluster:
+
+```
+kubectl apply -f catalogSource.yaml
+```
+
+This CatalogSource is created in an existing namespace by OLM named operators on OpenShift clusters. This namespace is called operators. When you create the CatalogSource, it causes the index image to be executed as a Pod. You can view it as follows:
+```
+kubectl -n operators get pod --selector=olm.catalogSource=sample-operator
+```
+
+#### Updating an index image
+
+After configuring OperatorHub to use a catalog source that references a custom index image that cluster administrators can keep the available Operators on their cluster up to date by adding bundle images to the index image.
+
+```
+opm index add --bundles <registry>/<namespace>/<new_bundle_image>:<tag> --from-index <registry>/<namespace>/<existing_index_image>:<tag> --tag <registry>/<namespace>/<existing_index_image>:<tag> 
+```
+
+- A comma seperated list of additional bundle images to the index
+- The existing index that was previously pushed
+- The image tag that you want the updated index image to have
+
+<br><br>
+For more references on Bundle Indices please read [OpenShift docs](https://docs.openshift.com/container-platform/4.6/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-bundle-format-prereqs) and this [blog](https://www.redhat.com/en/blog/deploying-operators-olm-bundles)
```

Comment:
It's weird to link to the OpenShift documentation from this upstream project. Are bundle indices not documented upstream?

Reply:
+1, we shouldn't have any mention of openshift at all in this docs page

Reply:
Please suggest any other source. I was not able to find any other good reference

Reply:
https://github.com/operator-framework/operator-registry#building-an-index-of-operators-using-opm